### ISSUE TITLE
UnescapeAttrVal

### ISSUE THREAD

**Issue by 89z** [tdewolff/parse#89]
I am looking for something to get attribute values without quotes. Currently I am using this:

~~~go
package parse

import (
   "bytes"
   "html"
)

func getAttr(b []byte) []byte {                                     
   b = bytes.Trim(b, `'"`)
   s := html.UnescapeString(string(b))
   return []byte(s)
}                                                                               
~~~

I found the `EscapeAttrVal` function [1]:

> EscapeAttrVal returns the escaped attribute value bytes without quotes. 

but I am confused. It says "without quotes", but it actually adds quotes:

https://github.com/tdewolff/parse/blob/f1ab946d07f836657ec47972229c77c0efa3b3bd/html/util_test.go#L17-L25

1. https://godocs.io/github.com/tdewolff/parse/v2/html#EscapeAttrVal

**Comment by tdewolff** [comment]
You're right that the comment is inaccurate. I've fixed the documentation, but note that the function has a different purpose than what you're looking for. If you want to remove quotes, you'll need to look if the first character is a single or double quote, and remove the first and last characters. The value may contain HTML entities (such as `&#39;`) which were originally present.

**Comment by 89z** [comment]
I might be wrong on this, but I think its not possible for an attribute to start with a literal single or double quote. If an attribute starts with one of those, it must be interpreted as a quote character, not as a literal character. For one of those to start the attribute literally, it would need to be encoded as an entity, like you mentioned.

If I am right, then the check for the first character would be redundant.
